provides.py

```diff
@@ -38,3 +38,10 @@ def set_dns_info(self, port, domain, sdn_ip):
                        'sdn-ip': sdn_ip}
         conv = self.conversation()
         conv.set_remote(data=credentials)
+
+    def set_seed(self, seed):
```

Comment:
I am trying to think of a better way to do this but at the moment I can not figure it out.